### PR TITLE
No-op if no client found in Faraday integration

### PR DESCRIPTION
**Pull request by Sergio-Mira**
https://github.com/honeycombio/beeline-ruby/issues/25

### CODE REVIEW

**Comment by Sergio-Mira** [comment]
(Looks like travis is missing the right ruby version `Required ruby-2.6.3 is not installed.`)

**Comment by martin308** [comment]
@Sergio-Mira @DanielHeath this should fix the problem here, and I'll get this merged. 

I would probably advise setting the client to be a `NullClient` though for your tests as otherwise you will not be exercising a whole heap of code paths.

```
Honeycomb.configure do |config|
  config.client = Libhoney:: NullClient.new
end
```

**Comment by martin308** [comment]
@Sergio-Mira added a test and a fix so that we continue to call any additional middleware that has been setup

File: lib/honeycomb/integrations/faraday.rb [hunk 1]
```diff
@@ -11,6 +11,8 @@ def initialize(app, client:)
     end
 
     def call(env)
+      return @app.call(env) if @client.nil?
```
Comment (Sergio-Mira) [inline lib/honeycomb/integrations/faraday.rb]: 👍

**Pull request merged**